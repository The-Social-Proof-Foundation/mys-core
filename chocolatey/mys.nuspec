<?xml version="1.0" encoding="utf-8"?>
<!-- Do not remove this test for UTF-8: if “Ω” doesn’t appear as greek uppercase omega letter
enclosed in quotation marks, you should use an editor that supports UTF-8, not this one. -->
<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
	<metadata>
		<id>mys</id>
		<version>$version$</version>
		<owners>MySocial Foundation</owners>
		<title>Main MySocial Binary</title>
		<authors>mys</authors>
		<projectUrl>https://mys.io/</projectUrl>
		<licenseUrl>https://github.com/MystenLabs/mys/blob/main/LICENSE</licenseUrl>
		<iconUrl>https://assets-global.website-files.com/6425f546844727ce5fb9e5ab/643775f4a15c9a9e10426daa_Mys_Favicon_256.png</iconUrl>
		<projectSourceUrl>https://github.com/MystenLabs/mys/</projectSourceUrl>
		<bugTrackerUrl>https://github.com/MystenLabs/mys/issues</bugTrackerUrl>
		<tags>mys</tags>
		<packageSourceUrl>https://community.chocolatey.org/packages/mys.portable</packageSourceUrl>
		<summary>MySocial delivers the benefits of Web3 with the ease of Web2</summary>
		<description>MySocial is the first internet-scale programmable blockchain platform</description>
		<releaseNotes>See https://github.com/MystenLabs/mys/releases/tag/testnet-v$version$</releaseNotes>
	</metadata>
	<files>
		<file src="../tmp/release/mys.exe" target="bin/mys.exe" />
		<file src="../LICENSE" target="bin/LICENSE.txt" />
		<file src="./VERIFICATION.txt" target="bin/VERIFICATION.txt" />
	</files>
</package>
