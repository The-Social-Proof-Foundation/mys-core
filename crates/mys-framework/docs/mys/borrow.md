---
title: Module `mys::borrow`
---

A simple library that enables hot-potato-locked borrow mechanics.

With Programmable transactions, it is possible to borrow a value within
a transaction, use it and put back in the end. Hot-potato <code><a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a></code> makes
sure the object is returned and was not swapped for another one.


-  [Struct `Referent`](#mys_borrow_Referent)
-  [Struct `Borrow`](#mys_borrow_Borrow)
-  [Constants](#@Constants_0)
-  [Function `new`](#mys_borrow_new)
-  [Function `borrow`](#mys_borrow_borrow)
-  [Function `put_back`](#mys_borrow_put_back)
-  [Function `destroy`](#mys_borrow_destroy)


<pre><code><b>use</b> <a href="../std/ascii.md#std_ascii">std::ascii</a>;
<b>use</b> <a href="../std/bcs.md#std_bcs">std::bcs</a>;
<b>use</b> <a href="../std/option.md#std_option">std::option</a>;
<b>use</b> <a href="../std/string.md#std_string">std::string</a>;
<b>use</b> <a href="../std/vector.md#std_vector">std::vector</a>;
<b>use</b> <a href="../mys/address.md#mys_address">mys::address</a>;
<b>use</b> <a href="../mys/hex.md#mys_hex">mys::hex</a>;
<b>use</b> <a href="../mys/object.md#mys_object">mys::object</a>;
<b>use</b> <a href="../mys/tx_context.md#mys_tx_context">mys::tx_context</a>;
</code></pre>



<a name="mys_borrow_Referent"></a>

## Struct `Referent`

An object wrapping a <code>T</code> and providing the borrow API.


<pre><code><b>public</b> <b>struct</b> <a href="../mys/borrow.md#mys_borrow_Referent">Referent</a>&lt;T: key, store&gt; <b>has</b> store
</code></pre>



<details>
<summary>Fields</summary>


<dl>
<dt>
<code>id: <b>address</b></code>
</dt>
<dd>
</dd>
<dt>
<code>value: <a href="../std/option.md#std_option_Option">std::option::Option</a>&lt;T&gt;</code>
</dt>
<dd>
</dd>
</dl>


</details>

<a name="mys_borrow_Borrow"></a>

## Struct `Borrow`

A hot potato making sure the object is put back once borrowed.


<pre><code><b>public</b> <b>struct</b> <a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a>
</code></pre>



<details>
<summary>Fields</summary>


<dl>
<dt>
<code>ref: <b>address</b></code>
</dt>
<dd>
</dd>
<dt>
<code>obj: <a href="../mys/object.md#mys_object_ID">mys::object::ID</a></code>
</dt>
<dd>
</dd>
</dl>


</details>

<a name="@Constants_0"></a>

## Constants


<a name="mys_borrow_EWrongBorrow"></a>

The <code><a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a></code> does not match the <code><a href="../mys/borrow.md#mys_borrow_Referent">Referent</a></code>.


<pre><code><b>const</b> <a href="../mys/borrow.md#mys_borrow_EWrongBorrow">EWrongBorrow</a>: u64 = 0;
</code></pre>



<a name="mys_borrow_EWrongValue"></a>

An attempt to swap the <code><a href="../mys/borrow.md#mys_borrow_Referent">Referent</a>.value</code> with another object of the same type.


<pre><code><b>const</b> <a href="../mys/borrow.md#mys_borrow_EWrongValue">EWrongValue</a>: u64 = 1;
</code></pre>



<a name="mys_borrow_new"></a>

## Function `new`

Create a new <code><a href="../mys/borrow.md#mys_borrow_Referent">Referent</a></code> struct


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow_new">new</a>&lt;T: key, store&gt;(value: T, ctx: &<b>mut</b> <a href="../mys/tx_context.md#mys_tx_context_TxContext">mys::tx_context::TxContext</a>): <a href="../mys/borrow.md#mys_borrow_Referent">mys::borrow::Referent</a>&lt;T&gt;
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow_new">new</a>&lt;T: key + store&gt;(value: T, ctx: &<b>mut</b> TxContext): <a href="../mys/borrow.md#mys_borrow_Referent">Referent</a>&lt;T&gt; {
    <a href="../mys/borrow.md#mys_borrow_Referent">Referent</a> {
        id: <a href="../mys/tx_context.md#mys_tx_context_fresh_object_address">tx_context::fresh_object_address</a>(ctx),
        value: option::some(value),
    }
}
</code></pre>



</details>

<a name="mys_borrow_borrow"></a>

## Function `borrow`

Borrow the <code>T</code> from the <code><a href="../mys/borrow.md#mys_borrow_Referent">Referent</a></code> receiving the <code>T</code> and a <code><a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a></code>
hot potato.


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow">borrow</a>&lt;T: key, store&gt;(self: &<b>mut</b> <a href="../mys/borrow.md#mys_borrow_Referent">mys::borrow::Referent</a>&lt;T&gt;): (T, <a href="../mys/borrow.md#mys_borrow_Borrow">mys::borrow::Borrow</a>)
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow">borrow</a>&lt;T: key + store&gt;(self: &<b>mut</b> <a href="../mys/borrow.md#mys_borrow_Referent">Referent</a>&lt;T&gt;): (T, <a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a>) {
    <b>let</b> value = self.value.extract();
    <b>let</b> id = <a href="../mys/object.md#mys_object_id">object::id</a>(&value);
    (
        value,
        <a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a> {
            ref: self.id,
            obj: id,
        },
    )
}
</code></pre>



</details>

<a name="mys_borrow_put_back"></a>

## Function `put_back`

Put an object and the <code><a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a></code> hot potato back.


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow_put_back">put_back</a>&lt;T: key, store&gt;(self: &<b>mut</b> <a href="../mys/borrow.md#mys_borrow_Referent">mys::borrow::Referent</a>&lt;T&gt;, value: T, <a href="../mys/borrow.md#mys_borrow">borrow</a>: <a href="../mys/borrow.md#mys_borrow_Borrow">mys::borrow::Borrow</a>)
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow_put_back">put_back</a>&lt;T: key + store&gt;(self: &<b>mut</b> <a href="../mys/borrow.md#mys_borrow_Referent">Referent</a>&lt;T&gt;, value: T, <a href="../mys/borrow.md#mys_borrow">borrow</a>: <a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a>) {
    <b>let</b> <a href="../mys/borrow.md#mys_borrow_Borrow">Borrow</a> { ref, obj } = <a href="../mys/borrow.md#mys_borrow">borrow</a>;
    <b>assert</b>!(<a href="../mys/object.md#mys_object_id">object::id</a>(&value) == obj, <a href="../mys/borrow.md#mys_borrow_EWrongValue">EWrongValue</a>);
    <b>assert</b>!(self.id == ref, <a href="../mys/borrow.md#mys_borrow_EWrongBorrow">EWrongBorrow</a>);
    self.value.fill(value);
}
</code></pre>



</details>

<a name="mys_borrow_destroy"></a>

## Function `destroy`

Unpack the <code><a href="../mys/borrow.md#mys_borrow_Referent">Referent</a></code> struct and return the value.


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow_destroy">destroy</a>&lt;T: key, store&gt;(self: <a href="../mys/borrow.md#mys_borrow_Referent">mys::borrow::Referent</a>&lt;T&gt;): T
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>public</b> <b>fun</b> <a href="../mys/borrow.md#mys_borrow_destroy">destroy</a>&lt;T: key + store&gt;(self: <a href="../mys/borrow.md#mys_borrow_Referent">Referent</a>&lt;T&gt;): T {
    <b>let</b> <a href="../mys/borrow.md#mys_borrow_Referent">Referent</a> { id: _, value } = self;
    value.destroy_some()
}
</code></pre>



</details>
