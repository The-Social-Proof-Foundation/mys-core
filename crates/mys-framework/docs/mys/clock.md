---
title: Module `mys::clock`
---

APIs for accessing time from move calls, via the <code><a href="../mys/clock.md#mys_clock_Clock">Clock</a></code>: a unique
shared object that is created at 0x6 during genesis.


-  [Struct `Clock`](#mys_clock_Clock)
-  [Constants](#@Constants_0)
-  [Function `timestamp_ms`](#mys_clock_timestamp_ms)
-  [Function `create`](#mys_clock_create)
-  [Function `consensus_commit_prologue`](#mys_clock_consensus_commit_prologue)


<pre><code><b>use</b> <a href="../std/ascii.md#std_ascii">std::ascii</a>;
<b>use</b> <a href="../std/bcs.md#std_bcs">std::bcs</a>;
<b>use</b> <a href="../std/option.md#std_option">std::option</a>;
<b>use</b> <a href="../std/string.md#std_string">std::string</a>;
<b>use</b> <a href="../std/vector.md#std_vector">std::vector</a>;
<b>use</b> <a href="../mys/address.md#mys_address">mys::address</a>;
<b>use</b> <a href="../mys/hex.md#mys_hex">mys::hex</a>;
<b>use</b> <a href="../mys/object.md#mys_object">mys::object</a>;
<b>use</b> <a href="../mys/transfer.md#mys_transfer">mys::transfer</a>;
<b>use</b> <a href="../mys/tx_context.md#mys_tx_context">mys::tx_context</a>;
</code></pre>



<a name="mys_clock_Clock"></a>

## Struct `Clock`

Singleton shared object that exposes time to Move calls.  This
object is found at address 0x6, and can only be read (accessed
via an immutable reference) by entry functions.

Entry Functions that attempt to accept <code><a href="../mys/clock.md#mys_clock_Clock">Clock</a></code> by mutable
reference or value will fail to verify, and honest validators
will not sign or execute transactions that use <code><a href="../mys/clock.md#mys_clock_Clock">Clock</a></code> as an
input parameter, unless it is passed by immutable reference.


<pre><code><b>public</b> <b>struct</b> <a href="../mys/clock.md#mys_clock_Clock">Clock</a> <b>has</b> key
</code></pre>



<details>
<summary>Fields</summary>


<dl>
<dt>
<code>id: <a href="../mys/object.md#mys_object_UID">mys::object::UID</a></code>
</dt>
<dd>
</dd>
<dt>
<code><a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>: u64</code>
</dt>
<dd>
 The clock's timestamp, which is set automatically by a
 system transaction every time consensus commits a
 schedule, or by <code>mys::clock::increment_for_testing</code> during
 testing.
</dd>
</dl>


</details>

<a name="@Constants_0"></a>

## Constants


<a name="mys_clock_ENotSystemAddress"></a>

Sender is not @0x0 the system address.


<pre><code><b>const</b> <a href="../mys/clock.md#mys_clock_ENotSystemAddress">ENotSystemAddress</a>: u64 = 0;
</code></pre>



<a name="mys_clock_timestamp_ms"></a>

## Function `timestamp_ms`

The <code><a href="../mys/clock.md#mys_clock">clock</a></code>'s current timestamp as a running total of
milliseconds since an arbitrary point in the past.


<pre><code><b>public</b> <b>fun</b> <a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>(<a href="../mys/clock.md#mys_clock">clock</a>: &<a href="../mys/clock.md#mys_clock_Clock">mys::clock::Clock</a>): u64
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>public</b> <b>fun</b> <a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>(<a href="../mys/clock.md#mys_clock">clock</a>: &<a href="../mys/clock.md#mys_clock_Clock">Clock</a>): u64 {
    <a href="../mys/clock.md#mys_clock">clock</a>.<a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>
}
</code></pre>



</details>

<a name="mys_clock_create"></a>

## Function `create`

Create and share the singleton Clock -- this function is
called exactly once, during genesis.


<pre><code><b>fun</b> <a href="../mys/clock.md#mys_clock_create">create</a>(ctx: &<a href="../mys/tx_context.md#mys_tx_context_TxContext">mys::tx_context::TxContext</a>)
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>fun</b> <a href="../mys/clock.md#mys_clock_create">create</a>(ctx: &TxContext) {
    <b>assert</b>!(ctx.sender() == @0x0, <a href="../mys/clock.md#mys_clock_ENotSystemAddress">ENotSystemAddress</a>);
    <a href="../mys/transfer.md#mys_transfer_share_object">transfer::share_object</a>(<a href="../mys/clock.md#mys_clock_Clock">Clock</a> {
        id: <a href="../mys/object.md#mys_object_clock">object::clock</a>(),
        // Initialised to zero, but set to a real timestamp by a
        // system transaction before it can be witnessed by a <b>move</b>
        // call.
        <a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>: 0,
    })
}
</code></pre>



</details>

<a name="mys_clock_consensus_commit_prologue"></a>

## Function `consensus_commit_prologue`



<pre><code><b>fun</b> <a href="../mys/clock.md#mys_clock_consensus_commit_prologue">consensus_commit_prologue</a>(<a href="../mys/clock.md#mys_clock">clock</a>: &<b>mut</b> <a href="../mys/clock.md#mys_clock_Clock">mys::clock::Clock</a>, <a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>: u64, ctx: &<a href="../mys/tx_context.md#mys_tx_context_TxContext">mys::tx_context::TxContext</a>)
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>fun</b> <a href="../mys/clock.md#mys_clock_consensus_commit_prologue">consensus_commit_prologue</a>(<a href="../mys/clock.md#mys_clock">clock</a>: &<b>mut</b> <a href="../mys/clock.md#mys_clock_Clock">Clock</a>, <a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>: u64, ctx: &TxContext) {
    // Validator will make a special system call with sender set <b>as</b> 0x0.
    <b>assert</b>!(ctx.sender() == @0x0, <a href="../mys/clock.md#mys_clock_ENotSystemAddress">ENotSystemAddress</a>);
    <a href="../mys/clock.md#mys_clock">clock</a>.<a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a> = <a href="../mys/clock.md#mys_clock_timestamp_ms">timestamp_ms</a>
}
</code></pre>



</details>
