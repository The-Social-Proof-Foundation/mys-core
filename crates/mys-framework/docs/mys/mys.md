---
title: Module `mys::mys`
---

Coin<MYS> is the token used to pay for gas in MySocial.
It has 9 decimals, and the smallest unit (10^-9) is called "mist".


-  [Struct `MYS`](#mys_mys_MYS)
-  [Constants](#@Constants_0)
-  [Function `new`](#mys_mys_new)
-  [Function `transfer`](#mys_mys_transfer)


<pre><code><b>use</b> <a href="../std/address.md#std_address">std::address</a>;
<b>use</b> <a href="../std/ascii.md#std_ascii">std::ascii</a>;
<b>use</b> <a href="../std/bcs.md#std_bcs">std::bcs</a>;
<b>use</b> <a href="../std/option.md#std_option">std::option</a>;
<b>use</b> <a href="../std/string.md#std_string">std::string</a>;
<b>use</b> <a href="../std/type_name.md#std_type_name">std::type_name</a>;
<b>use</b> <a href="../std/vector.md#std_vector">std::vector</a>;
<b>use</b> <a href="../mys/address.md#mys_address">mys::address</a>;
<b>use</b> <a href="../mys/bag.md#mys_bag">mys::bag</a>;
<b>use</b> <a href="../mys/balance.md#mys_balance">mys::balance</a>;
<b>use</b> <a href="../mys/coin.md#mys_coin">mys::coin</a>;
<b>use</b> <a href="../mys/config.md#mys_config">mys::config</a>;
<b>use</b> <a href="../mys/deny_list.md#mys_deny_list">mys::deny_list</a>;
<b>use</b> <a href="../mys/dynamic_field.md#mys_dynamic_field">mys::dynamic_field</a>;
<b>use</b> <a href="../mys/dynamic_object_field.md#mys_dynamic_object_field">mys::dynamic_object_field</a>;
<b>use</b> <a href="../mys/event.md#mys_event">mys::event</a>;
<b>use</b> <a href="../mys/hex.md#mys_hex">mys::hex</a>;
<b>use</b> <a href="../mys/object.md#mys_object">mys::object</a>;
<b>use</b> <a href="../mys/table.md#mys_table">mys::table</a>;
<b>use</b> <a href="../mys/transfer.md#mys_transfer">mys::transfer</a>;
<b>use</b> <a href="../mys/tx_context.md#mys_tx_context">mys::tx_context</a>;
<b>use</b> <a href="../mys/types.md#mys_types">mys::types</a>;
<b>use</b> <a href="../mys/url.md#mys_url">mys::url</a>;
<b>use</b> <a href="../mys/vec_set.md#mys_vec_set">mys::vec_set</a>;
</code></pre>



<a name="mys_mys_MYS"></a>

## Struct `MYS`

Name of the coin


<pre><code><b>public</b> <b>struct</b> <a href="../mys/mys.md#mys_mys_MYS">MYS</a> <b>has</b> drop
</code></pre>



<details>
<summary>Fields</summary>


<dl>
</dl>


</details>

<a name="@Constants_0"></a>

## Constants


<a name="mys_mys_EAlreadyMinted"></a>



<pre><code><b>const</b> <a href="../mys/mys.md#mys_mys_EAlreadyMinted">EAlreadyMinted</a>: u64 = 0;
</code></pre>



<a name="mys_mys_ENotSystemAddress"></a>

Sender is not @0x0 the system address.


<pre><code><b>const</b> <a href="../mys/mys.md#mys_mys_ENotSystemAddress">ENotSystemAddress</a>: u64 = 1;
</code></pre>



<a name="mys_mys_MIST_PER_MYS"></a>

The amount of Mist per MySocial token based on the fact that mist is
10^-9 of a MySocial token


<pre><code><b>const</b> <a href="../mys/mys.md#mys_mys_MIST_PER_MYS">MIST_PER_MYS</a>: u64 = 1000000000;
</code></pre>



<a name="mys_mys_TOTAL_SUPPLY_MIST"></a>

The total supply of MySocial denominated in Mist (10 Billion * 10^9)


<pre><code><b>const</b> <a href="../mys/mys.md#mys_mys_TOTAL_SUPPLY_MIST">TOTAL_SUPPLY_MIST</a>: u64 = 10000000000000000000;
</code></pre>



<a name="mys_mys_TOTAL_SUPPLY_MYS"></a>

The total supply of MySocial denominated in whole MySocial tokens (10 Billion)


<pre><code><b>const</b> <a href="../mys/mys.md#mys_mys_TOTAL_SUPPLY_MYS">TOTAL_SUPPLY_MYS</a>: u64 = 10000000000;
</code></pre>



<a name="mys_mys_new"></a>

## Function `new`

Register the <code><a href="../mys/mys.md#mys_mys_MYS">MYS</a></code> Coin to acquire its <code>Supply</code>.
This should be called only once during genesis creation.


<pre><code><b>fun</b> <a href="../mys/mys.md#mys_mys_new">new</a>(ctx: &<b>mut</b> <a href="../mys/tx_context.md#mys_tx_context_TxContext">mys::tx_context::TxContext</a>): <a href="../mys/balance.md#mys_balance_Balance">mys::balance::Balance</a>&lt;<a href="../mys/mys.md#mys_mys_MYS">mys::mys::MYS</a>&gt;
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>fun</b> <a href="../mys/mys.md#mys_mys_new">new</a>(ctx: &<b>mut</b> TxContext): Balance&lt;<a href="../mys/mys.md#mys_mys_MYS">MYS</a>&gt; {
    <b>assert</b>!(ctx.sender() == @0x0, <a href="../mys/mys.md#mys_mys_ENotSystemAddress">ENotSystemAddress</a>);
    <b>assert</b>!(ctx.epoch() == 0, <a href="../mys/mys.md#mys_mys_EAlreadyMinted">EAlreadyMinted</a>);
    <b>let</b> (treasury, metadata) = <a href="../mys/coin.md#mys_coin_create_currency">coin::create_currency</a>(
        <a href="../mys/mys.md#mys_mys_MYS">MYS</a> {},
        9,
        b"<a href="../mys/mys.md#mys_mys_MYS">MYS</a>",
        b"MySocial",
        // TODO: add appropriate description and logo <a href="../mys/url.md#mys_url">url</a>
        b"",
        option::none(),
        ctx,
    );
    <a href="../mys/transfer.md#mys_transfer_public_freeze_object">transfer::public_freeze_object</a>(metadata);
    <b>let</b> <b>mut</b> supply = treasury.treasury_into_supply();
    <b>let</b> total_mys = supply.increase_supply(<a href="../mys/mys.md#mys_mys_TOTAL_SUPPLY_MIST">TOTAL_SUPPLY_MIST</a>);
    supply.destroy_supply();
    total_mys
}
</code></pre>



</details>

<a name="mys_mys_transfer"></a>

## Function `transfer`



<pre><code><b>public</b> <b>entry</b> <b>fun</b> <a href="../mys/transfer.md#mys_transfer">transfer</a>(c: <a href="../mys/coin.md#mys_coin_Coin">mys::coin::Coin</a>&lt;<a href="../mys/mys.md#mys_mys_MYS">mys::mys::MYS</a>&gt;, recipient: <b>address</b>)
</code></pre>



<details>
<summary>Implementation</summary>


<pre><code><b>public</b> <b>entry</b> <b>fun</b> <a href="../mys/transfer.md#mys_transfer">transfer</a>(c: <a href="../mys/coin.md#mys_coin_Coin">coin::Coin</a>&lt;<a href="../mys/mys.md#mys_mys_MYS">MYS</a>&gt;, recipient: <b>address</b>) {
    <a href="../mys/transfer.md#mys_transfer_public_transfer">transfer::public_transfer</a>(c, recipient)
}
</code></pre>



</details>
