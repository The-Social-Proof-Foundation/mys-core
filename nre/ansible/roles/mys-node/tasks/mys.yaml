---
- name: Add a mys user
  ansible.builtin.user:
    name: mys
    shell: /bin/bash

- name: Create mys directories
  file:
    path: "{{ item }}"
    state: directory
    owner: "mys"
    group: "mys"
    mode: 0755
  loop:
    - /opt/mys/bin
    - /opt/mys/config
    - /opt/mys/db
    - /opt/mys/key-pairs

- name: Copy validator configuration file
  copy:
    src: "{{ node_config }}"
    dest: /opt/mys/config/validator.yaml
    owner: "mys"
    group: "mys"
    mode: 0644

- name: Copy genesis.blob
  copy:
    src: "{{ genesis_blob }}"
    dest: /opt/mys/config/genesis.blob
    owner: "mys"
    group: "mys"
    mode: 0644

- name: Copy protocol.key
  copy:
    src: "{{ keypair_path }}protocol.key"
    dest: /opt/mys/key-pairs/protocol.key
    owner: "mys"
    group: "mys"
    mode: 0600

- name: Copy network.key
  copy:
    src: "{{ keypair_path }}network.key"
    dest: /opt/mys/key-pairs/network.key
    owner: "mys"
    group: "mys"
    mode: 0600

- name: Copy worker.key
  copy:
    src: "{{ keypair_path }}worker.key"
    dest: /opt/mys/key-pairs/worker.key
    owner: "mys"
    group: "mys"
    mode: 0600

- name: Stop mys-node
  systemd:
    name: mys-node
    state: stopped
  ignore_errors: yes

- name: Remove authorities directory
  file:
    state: absent
    path: "/opt/mys/db/authorities_db"
  when: reset_db == true

- name: Remove consensus directory
  file:
    state: absent
    path: "/opt/mys/db/consensus_db"
  when: reset_db == true

- name: Place binary
  ansible.builtin.get_url:
    url: "https://releases.mys.io/{{ mys_release }}/mys-node"
    dest: /opt/mys/bin/mys-node
    force: true
    mode: "0755"

- name: Copy mys-node systemd service file
  copy:
    src: "../files/mys-node.service"
    dest: /etc/systemd/system/mys-node.service

- name: Reload mys-node systemd service file
  ansible.builtin.systemd:
    name: mys-node
    state: reloaded

- name: Start mys-node service
  systemd:
    name: mys-node
    state: started
  when: start_mys == true
