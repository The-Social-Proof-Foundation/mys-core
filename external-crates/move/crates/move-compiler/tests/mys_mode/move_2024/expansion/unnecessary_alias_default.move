module mys::object {
    public struct ID()
    public struct UID()
}
module mys::transfer {}
module mys::tx_context {
    public struct TxContext()
}

module a::m {
    use mys::object::{Self, ID, UID};
    use mys::transfer;
    use mys::tx_context::{Self, TxContext};
}
