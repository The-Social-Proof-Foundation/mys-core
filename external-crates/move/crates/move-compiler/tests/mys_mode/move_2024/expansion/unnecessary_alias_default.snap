---
source: crates/move-compiler/tests/move_check_testsuite.rs
info:
  flavor: mys
  edition: 2024.alpha
  lint: false
---
warning[W02021]: duplicate alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:11:23
   │
11 │     use mys::object::{Self, ID, UID};
   │                       ^^^^ Unnecessary alias 'object' for module 'mys::object'. This alias is provided by default
   │
   = This warning can be suppressed with '#[allow(duplicate_alias)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W09001]: unused alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:11:23
   │
11 │     use mys::object::{Self, ID, UID};
   │                       ^^^^ Unused 'use' of alias 'object'. Consider removing it
   │
   = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W02021]: duplicate alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:11:29
   │
11 │     use mys::object::{Self, ID, UID};
   │                             ^^ Unnecessary alias 'ID' for module member 'mys::object::ID'. This alias is provided by default
   │
   = This warning can be suppressed with '#[allow(duplicate_alias)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W09001]: unused alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:11:29
   │
11 │     use mys::object::{Self, ID, UID};
   │                             ^^ Unused 'use' of alias 'ID'. Consider removing it
   │
   = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W02021]: duplicate alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:11:33
   │
11 │     use mys::object::{Self, ID, UID};
   │                                 ^^^ Unnecessary alias 'UID' for module member 'mys::object::UID'. This alias is provided by default
   │
   = This warning can be suppressed with '#[allow(duplicate_alias)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W09001]: unused alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:11:33
   │
11 │     use mys::object::{Self, ID, UID};
   │                                 ^^^ Unused 'use' of alias 'UID'. Consider removing it
   │
   = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W02021]: duplicate alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:12:14
   │
12 │     use mys::transfer;
   │              ^^^^^^^^ Unnecessary alias 'transfer' for module 'mys::transfer'. This alias is provided by default
   │
   = This warning can be suppressed with '#[allow(duplicate_alias)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W09001]: unused alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:12:14
   │
12 │     use mys::transfer;
   │              ^^^^^^^^ Unused 'use' of alias 'transfer'. Consider removing it
   │
   = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W02021]: duplicate alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:13:27
   │
13 │     use mys::tx_context::{Self, TxContext};
   │                           ^^^^ Unnecessary alias 'tx_context' for module 'mys::tx_context'. This alias is provided by default
   │
   = This warning can be suppressed with '#[allow(duplicate_alias)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W09001]: unused alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:13:27
   │
13 │     use mys::tx_context::{Self, TxContext};
   │                           ^^^^ Unused 'use' of alias 'tx_context'. Consider removing it
   │
   = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W02021]: duplicate alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:13:33
   │
13 │     use mys::tx_context::{Self, TxContext};
   │                                 ^^^^^^^^^ Unnecessary alias 'TxContext' for module member 'mys::tx_context::TxContext'. This alias is provided by default
   │
   = This warning can be suppressed with '#[allow(duplicate_alias)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

warning[W09001]: unused alias
   ┌─ tests/mys_mode/move_2024/expansion/unnecessary_alias_default.move:13:33
   │
13 │     use mys::tx_context::{Self, TxContext};
   │                                 ^^^^^^^^^ Unused 'use' of alias 'TxContext'. Consider removing it
   │
   = This warning can be suppressed with '#[allow(unused_use)]' applied to the 'module' or module member ('const', 'fun', or 'struct')
